siobrultech_protocols/gem/protocol.py

```diff
@@ -15,16 +16,58 @@
 LOG = logging.getLogger(__name__)
 
 PACKET_HEADER = bytes.fromhex("feff")
+API_RESPONSE_WAIT_TIME_SECONDS = 3.0  # Time to wait for an API response
+PACKET_DELAY_CLEAR_TIME_SECONDS = 3.0  # Time to wait after a PDL request so that GEM can finish sending any pending packets
 
 
 class PacketProtocol(asyncio.Protocol):
-    def __init__(self, queue: asyncio.Queue):
+    def __init__(
+        self,
+        queue: asyncio.Queue,
+        on_connection_made: Optional[Callable[[GemApi], NoReturn]] = None,
+    ):
         self._buffer = bytearray()
         self._queue = queue
-        self._transport: Optional[asyncio.BaseTransport] = None
-
-    def connection_made(self, transport: asyncio.BaseTransport):
+        self._transport: Optional[asyncio.WriteTransport] = None
+        self._api_lock = asyncio.Lock()
+        self._api_mode = False
+        self._on_connection_made = on_connection_made
+
+    async def _send_api_command(self, command: str):
+        with await self._api_lock:  # One API call at a time, please
+            if not self._transport:
+                raise EOFError
+
+            # We're about to send a request on the same channel that the GEM is using to
+            # push data packets to us. To minimize confusion, we ask GEM to delay packets
+            # for 15 seconds and give it a few seconds to finish sending any in-progress
+            # packets before sending our request.
+            self._transport.write("^^^SYSPDL".encode())  # Delay packets for 15 seconds
```

Comment:
I feel like the `^^^` should probably be the packet header for commands, and therefore a `const`.  It might be good to makeup `const`s for the commands to so they are more readable as we add more.  While I can constantly reference the docs, it'd be nice to have descriptive names of the command instead :D